### DISCUSSION TITLE
Using NGINX reverse proxy with subpath gives "404: Not Found"

### DISCUSSION THREAD

**Discussion by g4njawizard** [dani-garcia/vaultwarden#2289] [Q&A] [answered]
Hi everyone,

I try to setup Vaultwarden on my unRAID as container using NGINX Reverse Proxy on my OPNsense Firewall (Different machine).

It works with my current proxy setting, when I use "/" as base path. But when I change my path to "/bitwarden/asgard" or something else and change the domain on the container itself to look like: -e 'DOMAIN'='http://localhost/bitwarden/asgard' It doesn't work. I get the 404.
I can leave the Domain on the Container with "http://localhost/bitwarden/asgard" and change the Proxy config to "/" again, it works ... 
Seems like the base path inside the container is not working? When I look at the read-only config in admin section it looks like:
Domain Origin:  http://localhost
Domain Path: Default: /bitwarden/asgard
I used this config as an example: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

Any idea why subpath is not working?

**Comment by BlackDex** [comment]
Make sure you have a trailing `/` in your nginx config for the main path, so for the `location /bitwarden/asgard/ {`, without the trailing slash in nginx you could experience issues. For the Vaultwarden `DOMAIN` you can leave the trailing slash.

**Comment by g4njawizard** [comment]
I got that. Saw that in the wiki, that trailing slash could be an issue. I tried it in various ways with URL rewriting, that it adds the slash in the end of /bitwarden/asgard, but that all doesnt help. 
I have the feeling that my proxy config might be fine and Rocket might be causing this issue:

404: Not Found

The requested resource could not be found.
Rocket 

As soon as I revert it the proxy to "/" only it works... is there a log inside the container that shows what URL the user tries to access?

**Comment by g4njawizard** [comment]
This is the part describing this issue:

```

If you normally configure vaultwarden via environment variables, update your config files/scripts to set the DOMAIN environment variable to the base URL. For example:

docker run -e DOMAIN="https://vaultwarden.example.com/base-dir" ...

Restart vaultwarden.

You should now be able to access the web vault at https://vaultwarden.example.com/base-dir/ (note the trailing slash). For reasons not entirely clear, you'll probably run into issues if you use https://vaultwarden.example.com/base-dir (without the trailing slash).

Configure your apps or browser extensions to use https://vaultwarden.example.com/base-dir. If you add a trailing slash, the apps and extensions will automatically remove it before saving.

Note over 5. The trailing slash / issue could be solved by appending / after the route location string. For example, in nginx.

location /my-base-path {
  # This config would cause `/` issue
}

location /my-base-path-2/ {
  # This config works perfectly
}
```
mine in nginx is: /bitwarden/asgard/ and I've set it DOMAIN=https://vault.example.com/bitwarden/asgard in the container..
I cannot find logs or something where I can peak .. Hope that I dont have to use wireshark.

I checked the Networkanalyse Tool from Firefox.
It seems that after I try to reach: https://vault.example.com/bitwarden/asgard/
The Second Get which throws 404 looks: https://vault.example.com/favicon.ico

It seems that the path is being ignored. Need to check if thats nginx fault or not..

**Comment by g4njawizard** [comment, marked as answer]
404 is gone.
Seems that in my case, I have to set the "Path Prefix" to "/" inside the location /bitwarden/asgard/ Maybe this could be mentioned in the wiki as sidenote for OPNsense users idk.

**Reply by BlackDex** [reply]
What do you mean with "Path Prefix" exactly?

**Reply by g4njawizard** [reply]
I cant explain what it exactly is. It appends the trailing slash in the proxy_pass
I just found this in here https://forum.opnsense.org/index.php?topic=20943.msg98074#msg98074  not related to vaultwarden
There are screenshots, but only visible to members.
